Ankidroid Decimal Intervals under 4 secs Not Allowed - Giving Error

###### Reproduction Steps
UPDATE: It seems the error is for steps lower than 4secs (=0.067mins), this is the reason <0.067min intervals (like 0.05 0.02) give error

1. Hold click a deck and choose Options
2. go to New cards or Lapses
3. Click "Steps in Minutes"
4. add an interval with decimals f.e. 1.5 or 0.5 or 0.02


###### Expected Result
To show the card and be able to select easy/hard etc and show the next card

As in Anki PC it allows you to set intervals with decimal spaces, which is very useful especially in hard cards


###### Actual Result
When I open a deck with settings with decimals (f.e. one that synced from pc), it shows the card ok, but when I tap an option "easy" "hard" etc which though includes decimals in scheduling f.e. 0.1min, 1.5min then it sends an error as shown in the picture saying: "Collection not opened" an error occurred while accessing your collection... 

Why I know it is the decimals? If I set the intervals in integers f.e. 1min 5min 2min, then everything is ok.

![image](https://user-images.githubusercontent.com/1826311/102781475-68fa6200-43a0-11eb-95be-a92698cec40d.png)

###### Does it happen again every time you repeat the steps above? Or did it
happen only one time?
Every time

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

AnkiDroid Version = 2.14.2

Android Version = 9

ACRA UUID = dc880539-d555-4065-9ea9-b0d24e7f1564

Scheduler = std2

Crash Reports Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum

PLEASE fix soon so I can study from my phone optimized. thanks!!!

Reproducible on SchedV2

```
2021-01-27 15:27:10.882 7347-9008/com.ichi2.anki E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #19
    Process: com.ichi2.anki, PID: 7347
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:355)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:246)
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:784)
     Caused by: java.lang.IllegalArgumentException: bound must be positive
        at java.util.Random.nextInt(Random.java:388)
        at com.ichi2.libanki.sched.SchedV2._rescheduleLrnCard(SchedV2.java:1271)
        at com.ichi2.libanki.sched.SchedV2._rescheduleLrnCard(SchedV2.java:1256)
        at com.ichi2.libanki.sched.SchedV2._moveToFirstStep(SchedV2.java:1236)
        at com.ichi2.libanki.sched.SchedV2._answerLrnCard(SchedV2.java:1216)
        at com.ichi2.libanki.sched.SchedV2._answerCard(SchedV2.java:316)
        at com.ichi2.libanki.sched.SchedV2.answerCard(SchedV2.java:289)
        at com.ichi2.async.CollectionTask$AnswerAndGetCard.task(CollectionTask.java:368)
        at com.ichi2.async.CollectionTask$AnswerAndGetCard.task(CollectionTask.java:358)
        at com.ichi2.async.CollectionTask.actualDoInBackground(CollectionTask.java:222)
        at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:185)
        at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:108)
        at android.os.AsyncTask$2.call(AsyncTask.java:334)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:246) 
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:784) 
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.java
@@ -1268,7 +1268,7 @@ public class SchedV2 extends AbstractSched {
         if (card.getDue() < mDayCutoff) {
             // Add some randomness, up to 5 minutes or 25%
             int maxExtra = Math.min(300, (int)(delay * 0.25));
-            int fuzz = new Random().nextInt(maxExtra);
+            int fuzz = new Random().nextInt(Math.max(maxExtra, 1));
             card.setDue(Math.min(mDayCutoff - 1, card.getDue() + fuzz));
             card.setQueue(Consts.QUEUE_TYPE_LRN);
             if (card.getDue() < (getTime().intTime() + mCol.getConf().getInt("collapseTime"))) {

--- a/AnkiDroid/src/test/java/com/ichi2/libanki/sched/SchedV2Test.java
+++ b/AnkiDroid/src/test/java/com/ichi2/libanki/sched/SchedV2Test.java
@@ -75,6 +75,7 @@ import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.lessThan;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
 import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotEquals;
@@ -255,6 +256,22 @@ public class SchedV2Test extends RobolectricTest {
 
         assertThat("Card should be moved to the home deck", getCol().getCard(cardId).getDid(), is(1L));
         assertThat("Card should not be in a filtered deck", getCol().getCard(cardId).getODid(), is(0L));
+    }
+
+    @Test
+    public void handlesSmallSteps() throws ConfirmModSchemaException {
+        // a delay of 0 crashed the app (step of 0.01).
+        getCol().changeSchedulerVer(2);
+
+        addNoteUsingBasicModel("Hello", "World");
+
+        getCol().getDecks().allConf().get(0).getJSONObject("new").put("delays", new JSONArray(Arrays.asList(0.01, 10)));
+
+        Card c = getCol().getSched().getCard();
+
+        assertThat(c, notNullValue());
+
+        getCol().getSched().answerCard(c, 1);
     }
 
 

